rule-types/common/enforce_file.yaml

```
@@ -0,0 +1,89 @@
+---
+version: v1
+release_phase: alpha
+type: rule-type
+name: enforce_file
+display_name: Enforce a file in the repository
+short_failure_message: File does not exist or does not match the expected content
+severity:
+  value: medium
+context: {}
+description: |
+  Enforce the presence of a file and its content in the repository.
+guidance: |
+  Ensures that the repository contains a specific file and its content.
+  
+  This rule is useful for enforcing the presence of various files in the repository, such as LICENSE, README, CONTRIBUTING,
+  CODE_OF_CONDUCT, Dependabot configuration files and many more. It can also be used to enforce the presence of specific
+  content in the file so you know that the file is not just a placeholder.
+def:
+  in_entity: repository
+  rule_schema:
+    type: object
+    properties:
+      file:
+        type: string
+        description: |
+          The file to enforce in the repository.
+            For example, LICENSE, README, CONTRIBUTING, .github/dependabot.yml, etc.
+        default: LICENSE
+      content:
+        type: string
+        description: |
+          The content to enforce in the file.
+          For example, the content of the LICENSE file.
+          
+          Leave "" to only check for the presence of the license file.
+    required:
+      - file
+      - content
```

Question: right now, it looks like you'd end up needing to specify N copies of the rule for N different files.

What would you think of having a schema like the following?

```suggestion
      files:
        type: object
        additionalProperties:
          type: string
        description: |
          A mapping from file path to expected contents in the repository
          
          Set expected contents to "" (empty string) to check for file existence without
          checking file contents.
        default: 
    required:
      - files
```